## V katerem področju se zgodi največ nesreč, in kje se zgodi največ nesreč povezanih z alkoholom

### X in Y koordinate bom razdelil v meje, vsebovane v kvadratkih, ki jih bom spodaj izračunal

In [30]:
# Reevantni importi
% matplotlib inline
import matplotlib.image as mpimg
import matplotlib.pyplot as plt
from Orange.data.filter import SameValue
from Orange.data import Table
import numpy as np
from csv import DictReader

In [31]:
# Filtrirani podatki naloženi (crash data filt je isti, ker je tako ali tako cel)
crash1_reader = DictReader(open('data/Crash_Data_1_filt.csv', 'rt', encoding='utf-8'))
crash2_reader = DictReader(open('data/Crash_Data_2_filt_non_rand.csv', 'rt', encoding='utf-8'))
liquor_reader = DictReader(open('data/Liquor_Data_filt_non_rand.csv', 'rt', encoding='utf-8'))

In [32]:
# Pregledam kako vnosi podatkov zgledajo
for crash in crash1_reader:
    print(crash)
    break
print("------------------")
for crash in crash2_reader:
    print(crash)
    break
print("------------------")
for liquor in liquor_reader:
    print(liquor)
    break

OrderedDict([('\ufeffX', '-93.511904910700665'), ('Y', '41.648063161780378'), ('OBJECTID', '1001'), ('CRASH_KEY', '2016902721'), ('CASENUMBER', '2016902721'), ('LECASENUM', '16-452'), ('CRASH_DATE', '2016-01-20T00:00:00.000Z'), ('CRASH_MONTH', '1'), ('CRASH_DAY', '4'), ('TIMESTR', '20:02'), ('DISTRICT', '1'), ('COUNTY_NUMBER', '77'), ('CITYNAME', '132'), ('SYSTEMSTR', 'US 6'), ('LITERAL', 'US 6/HUBBELL AVE/NE RAMP/SPEC CASE/FREDERICK M HUBBELL AVE'), ('FRSTHARM', '33'), ('LOCFSTHRM', '1'), ('CRCOMNNR', '4'), ('MAJCSE', '9'), ('DRUGALC', '8'), ('ECNTCRC', '1'), ('LIGHT', '5'), ('CSRFCND', '1'), ('WEATHER', '1'), ('RCNTCRC', '1'), ('RDTYP', '22'), ('PAVED', '1'), ('WZRELATED', ''), ('CSEV', '5'), ('FATALITIES', '0'), ('INJURIES', '0'), ('MAJINJURY', '0'), ('MININJURY', '0'), ('POSSINJURY', '0'), ('UNKINJURY', '0'), ('PROPDMG', '6000'), ('VEHICLES', '2'), ('TOCCUPANTS', '2'), ('REPORT', '7'), ('XCOORD', '457373'), ('YCOORD', '4610828')])
------------------
OrderedDict([('\ufeffX', '-92.65

### Spodaj bom izračunal meje kvadratkov koordinatnega sistema, s pomočjo minimalnih in maksimalnih koordinatov v podatkih

In [33]:
# Število X Koordinate (UTM NAD 83 Zone 15)
# Število Y Koordinate (UTM NAD 83 Zone 15)
# 
# Računa
crash1_max_X = -1000;
crash1_max_Y = -1000;
crash1_min_X = 1000;
crash1_min_Y = 1000;
for crash in crash1_reader:
    if float(crash["\ufeffX"])>crash1_max_X:
        crash1_max_X=float(crash["\ufeffX"])
    if float(crash["\ufeffX"])<crash1_min_X:
        crash1_min_X=float(crash["\ufeffX"])
    if float(crash["Y"])>crash1_max_Y:
        crash1_max_Y=float(crash["Y"])
    if float(crash["Y"])<crash1_min_Y:
        crash1_min_Y=float(crash["Y"])
print("------------------")
crash2_max_X = -1000;
crash2_max_Y = -1000;
crash2_min_X = 1000;
crash2_min_Y = 1000;
for crash in crash2_reader:
    if float(crash["\ufeffX"])>crash2_max_X:
        crash2_max_X=float(crash["\ufeffX"])
    if float(crash["\ufeffX"])<crash2_min_X:
        crash2_min_X=float(crash["\ufeffX"])
    if float(crash["Y"])>crash2_max_Y:
        crash2_max_Y=float(crash["Y"])
    if float(crash["Y"])<crash2_min_Y:
        crash2_min_Y=float(crash["Y"])
print("Data 1:")
print(str(crash1_min_X)+" , "+str(crash1_max_Y)+"--------"+str(crash1_max_X)+" , "+str(crash1_max_Y))
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print(str(crash1_min_X)+" , "+str(crash1_min_Y)+"--------"+str(crash1_max_X)+" , "+str(crash1_min_Y))
print("Data 2:")
print(str(crash2_min_X)+" , "+str(crash2_max_Y)+"--------"+str(crash2_max_X)+" , "+str(crash2_max_Y))
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print("---------------------------------------------------------------------------------")
print(str(crash2_min_X)+" , "+str(crash2_min_Y)+"--------"+str(crash2_max_X)+" , "+str(crash2_min_Y))

------------------
Data 1:
-96.64370824827424 , 43.50108618320072---------92.4040052229149 , 43.50108618320072
---------------------------------------------------------------------------------
---------------------------------------------------------------------------------
---------------------------------------------------------------------------------
---------------------------------------------------------------------------------
---------------------------------------------------------------------------------
---------------------------------------------------------------------------------
-96.64370824827424 , 40.2330982959834---------92.4040052229149 , 40.2330982959834
Data 2:
-96.64370824827424 , 43.50100896123742---------92.4040052229149 , 43.50100896123742
---------------------------------------------------------------------------------
---------------------------------------------------------------------------------
-----------------------------------------------------------

**Podatki za meje koordinatov so torej sledeči: **

A: 43.5010861832007 , -96.64370824827424   B:  43.50108618320072 , -92.4040052229149


C: 40.2330982959834 , -96.64370824827424    D: 40.2330982959834 , -92.4040052229149



#### Prek google maps sem dobil razdalje od točke do točke : 
###### A - B je 370 kilometrov
###### A - C je 450 kilometrov
###### C - D je 434 kilometrov
###### B - D je 410 kilometrov

In [34]:
difference_crash_X = crash1_max_X - crash1_min_X
difference_crash_Y = crash1_max_Y - crash1_min_Y
print(str(difference_crash_X)+" in "+str(difference_crash_Y))

4.239703025359333 in 3.2679878872173163


In [37]:
# Razlika med max X in min X, ter razlika med max Y in min Y: 4.239703025359333 in 3.2679878872173163
crash_x_limit = difference_crash_X/200
crash_y_limit = difference_crash_Y/200
print(crash_x_limit)
print(crash_y_limit)

0.021198515126796665
0.01633993943608658
